### ISSUE TITLE
OpenTelemetry.Instrumentation.Runtime: start error method not found ObservableUpDownCounter

### ISSUE THREAD

**Issue by Abdulsamirkhan** [open-telemetry/opentelemetry-dotnet-instrumentation#3154]
on startup function started getting below error without any changes to code just after new build with existing code(was working fine before the new build)
Microsoft.Azure.WebJobs.Script: Error configuring services in an external startup class. OpenTelemetry.Instrumentation.Runtime: The type initializer for 'OpenTelemetry.Instrumentation.Runtime.RuntimeMetrics' threw an exception. OpenTelemetry.Instrumentation.Runtime: Method not found: 'System.Diagnostics.Metrics.ObservableUpDownCounter`1<!!0> System.Diagnostics.Metrics.Meter.CreateObservableUpDownCounter(System.String, System.Func`1<!!0>, System.String, System.String)'.

- OpenTelemetry Automatic Instrumentation version: 1.5.1
- Azure function runtime 4
-  .NET version: 6.0

Detailed error : 

Microsoft.Azure.WebJobs.Script.ExternalStartupException : Error configuring services in an external startup class. ---> System.TypeInitializationException : The type initializer for 'OpenTelemetry.Instrumentation.Runtime.RuntimeMetrics' threw an exception. ---> System.MissingMethodException : Method not found: 'System.Diagnostics.Metrics.ObservableUpDownCounter`1<!!0> System.Diagnostics.Metrics.Meter.CreateObservableUpDownCounter(System.String, System.Func`1<!!0>, System.String, System.String)'. at OpenTelemetry.Instrumentation.Runtime.RuntimeMetrics..cctor() End of inner exception at OpenTelemetry.Metrics.MeterProviderBuilderExtensions.AddRuntimeInstrumentation(MeterProviderBuilder builder,Action`1 configure)

**Comment by RassK** [comment]
[CreateObservableUpDownCounter](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.metrics.meter.createobservableupdowncounter?view=net-7.0) seems to be supported from `System.Diagnostics.DiagnosticSource` `7.0.0`

@Abdulsamirkhan can you manually update the `System.Diagnostics.DiagnosticSource` >=  `7.0.0` and try again?

**Comment by RassK** [comment]
@Abdulsamirkhan are you using in-process azure functions? If I remember correctly .NET 6 and in-process azure functions have conflicts with some framework shipped libraries, like `System.Diagnostics.DiagnosticSource`.

**Comment by Abdulsamirkhan** [comment]
@RassK  we are not using in process azure functions, and same code is working fine till Friday and we release new build on Tuesday started causing this issue

**Comment by Kielek** [comment]
Do you really using [OpenTelemetry.AutoInstrumentation](https://www.nuget.org/packages/OpenTelemetry.AutoInstrumentation)? nuget package? Please list all packages used by your application.

[Minimal, Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example) with exact steps to reproduce will be also helpful.

**Comment by mada-vidyasagar** [comment]
No, We are not using OpenTelemetry.AutoInstrumentation package.
Following are the opentelemetry packages used in our application.

<PackageReference Include="OpenTelemetry" Version="1.6.0" />
<PackageReference Include="OpenTelemetry.Exporter.OpenTelemetryProtocol" Version="1.6.0" />
<PackageReference Include="OpenTelemetry.Extensions.Hosting" Version="1.6.0" />
<PackageReference Include="OpenTelemetry.Instrumentation.AspNetCore" Version="1.6.0-beta.3" />
<PackageReference Include="OpenTelemetry.Instrumentation.Runtime" Version="1.5.1" />
<PackageReference Include="System.Diagnostics.DiagnosticSource" Version="7.0.2" />
